### ISSUE TITLE
[Feature request] Nickname registration

### ISSUE THREAD

**Issue by finalhit** [kiwiirc/kiwiirc#1097]
It would be nice to have nickname registration system.

It is just too complicated for novice users to use /REGISTER and /NS CONFIRM commands.

Or even better is to support multiple providers like FirebaseUI, to allow sign-in using email/password, email link, phone authentication, Google Sign-In, Facebook Login, Twitter Login, GitHub Login, etc:
https://firebase.google.com/docs/auth/web/firebaseui

**Comment by Fudster** [comment]
This is more of a plugin, in my opinnon. 

https://github.com/insomniac807/kiwiplugins is a example. But I am not sure if it works.

@ItsOnlyBinary HAD a plugin for registering? I'm not sure... I have used a nickserv KiwiIRC plugin.

**Comment by Fudster** [comment]
I mentioned your issue on IRC here's another one: https://github.com/SimosNap/kiwiirc-plugin-nickserv

 nickserv plugin is for anope, but you have to  adapt regex to match your service https://github.com/SimosNap/kiwiirc-plugin-nickserv

**Comment by finalhit** [comment]
https://github.com/SimosNap/kiwiirc-plugin-nickserv is interesting.

It communicates with Anope web configuration panel. Not very fond of it, because it does not support reCAPTCHA nor any other method against brute force attacks.

Also, I think registration dialog should be displayed right at Welcome screen, before entering IRC.

Ideally user would be able to authenticate using email/password, phone authentication, Google Sign-In, Facebook Login, Twitter Login, etc via FirebaseUI or similar service. Then would need to sync that database with internal database accessible from Anope:
https://wiki.anope.org/index.php/2.0/Modules/m_sql_authentication

**Comment by prawnsalad** [comment]
This is something we want to put into kiwi itself but we're waiting on some IRCv3 standard features to be finalised yet. This would mean we can make it work on any IRC server without special weird edge cases.

Closing for now, but it's definitely coming in the future.
